Tools/build/check_warnings.py

```diff
@@ -32,7 +34,7 @@ def extract_warnings_from_compiler_output_clang(
         if match := clang_warning_regex.match(line):
             compiler_warnings.append(
                 {
-                    "file": match.group("file"),
+                    "file": match.group("file").lstrip(path_prefix),
```

Comment:
`.lstrip()` doesn't work how you'd expect (instead removing characters passed individually, not as a unit), we should use `.removeprefix()` instead.

Reply:
Wow very interesting I read the docs and learned something! Changed to `.removeprefix()`